test/rules/require_trailing_comma.dart

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+class RequireTrailingCommaExample {
+  RequireTrailingCommaExample.constructor1(Object param1, Object param2);
+
+  RequireTrailingCommaExample.constructor2(
+      Object param1,
+      Object param2,
+      Object param3,
+      );
+
+  RequireTrailingCommaExample.constructor3(
+      Object param1, Object param2, Object param3); // LINT
+
+  RequireTrailingCommaExample.constructor4(
+      Object param1,
+      Object param2, [
+        Object param3 = const [
+          'test',
+        ],
+      ]);
+
+  RequireTrailingCommaExample.constructor5(Object param1, Object param2,
+      [Object param3 = const [
+        'test',
+      ]]); // LINT
+
+  void method1(Object param1, Object param2, {Object param3, Object param4}) {}
+
+  void method2(
+      Object param1,
+      Object param2,
+      Object param3,
+      Object param4,
+      Object param5,
+      ) {}
+
+  void method3(
```

Comment:
I might have missed it, but I don't see an example of a named parameter with a default value.

Reply:
Added test case for named param with default value.